applications/luci-app-libreswan/htdocs/luci-static/resources/view/libreswan/tunnels.js

```diff
@@ -0,0 +1,241 @@
+		o.datatype = 'string';
+		o.multiple = false;
+		o.optional = true;
+
+		o = s.taboption('general', form.Value, 'left', _('Left IP/Device'));
+		o.datatype = 'or(string, ipaddr)';
+		for (var i = 0; i < netDevs.length; i++) {
+			var addrs = netDevs[i].getIPAddrs();
+			for (var j = 0; j < addrs.length; j++) {
+				o.value(addrs[j].split('/')[0]);
+			}
+		}
+		for (var i = 0; i < netDevs.length; i++) {
+			o.value('%' + netDevs[i].device);
+		}
+		o.value('%defaultroute');
+		o.optional = false;
+		o.depends({ 'left_interface' : '' });
+
+		o = s.taboption('general', form.Value, 'leftid', _('Left ID'));
+		o.datatype = 'string';
+		o.value('%any');
+		o.modalonly = true;
+
+		o = s.taboption('general', form.Value, 'right', _('Remote IP'));
+		o.datatype = 'or(string, ipaddr)';
+		o.value('0.0.0.0');
+		o.value('%any');
+		o.optional = false;
+
+		o = s.taboption('general', form.Value, 'rightid', _('Right ID'));
+		o.datatype = 'string';
+		o.value('%any');
+		o.modalonly = true;
+
+		o = s.taboption('general', form.Value, 'leftsourceip', _('Local Source IP'));
+		o.datatype = 'ipaddr';
+		for (var i = 0; i < netDevs.length; i++) {
+			var addrs = netDevs[i].getIPAddrs();
+			for (var j = 0; j < addrs.length; j++) {
+				o.value(addrs[j].split('/')[0]);
+			}
+		}
+		o.optional = false;
+		o.modalonly = true;
+
+		o = s.taboption('general', form.Value, 'rightsourceip', _('Remote Source IP'));
+		o.datatype = 'ipaddr';
+		o.optional = false;
+		o.modalonly = true;
+
+		o = s.taboption('general', form.DynamicList, 'leftsubnets', _('Local Subnets'));
+		o.datatype = 'ipaddr';
+		for (var i = 0; i < netDevs.length; i++) {
+			var addrs = netDevs[i].getIPAddrs();
+			for (var j = 0; j < addrs.length; j++) {
+				var subnet = calculateNetwork(addrs[j].split('/')[0], addrs[j].split('/')[1]);
+				if (subnet) {
+					o.value(subnet);
+				}
+			}
+		}
+		o.value('0.0.0.0/0');
+
+		o = s.taboption('general', form.DynamicList, 'rightsubnets', _('Remote Subnets'));
+		o.datatype = 'ipaddr';
+		o.value('0.0.0.0/0');
+
+		o = s.taboption('authentication', form.ListValue, 'authby', _('Auth Method'));
+		o.default = 'secret'
+		o.value('secret', 'Preshare Key');
+		o.modalonly = true;
+		o.optional = false;
+
+		o = s.taboption('authentication', form.Value, 'psk', _('Preshare Key'));
+		o.datatype = 'and(string, minlenght(8))'
+		o.password = true;
+		o.modalonly = true;
+		o.optional = false;
+
+		o = s.taboption('advanced', form.ListValue, 'ikev2', _('IKE V2'));
+		o.default = 'yes';
+		o.value('yes', _('IKE Version 2'));
+		o.value('no', _('IKE Version 1'));
+		o.modalonly = true;
+
+		o = s.taboption('advanced', form.MultiValue, 'ike', _('Phase1 Proposals'));
+		for (var i = 0; i < proposals.length; i++) {
+			o.value(proposals[i]['.name']);
+		}
+		o.modalonly = true;
+
+		o = s.taboption('advanced', form.Value, 'ikelifetime', _('IKE Life Time'));
+		o.datatype = 'uinteger';
+		o.default = 10800;
```

Comment:
Please add multiple values to choose from (and hint in readable text the amount of hours it represents), and permit the user to enter their own values. 

If the current libreswan version is >=4.2, then the default is `8h`. 

I do not think that this should be an unsigned integer...

Reply:
Suggestions: 1h, 2h, 4h, 8h, 12h, 16, 24h, custom...

Reply:
Updated